File: Muc_Do_4_Rat_Kho/Bai20_XoaPhanTu/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI20.INP', 'r')
        sys.stdout = open('BAI20.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    arr = [int(x) for x in data[1:1+n]]
    
    even_tot = sum(arr[i] for i in range(0, n, 2))
    odd_tot = sum(arr[i] for i in range(1, n, 2))
    
    ans = 0
    even_left = odd_left = 0
    for i in range(n):
        if i % 2 == 0:
            even_right = even_tot - even_left - arr[i]
            odd_right = odd_tot - odd_left
        else:
            even_right = even_tot - even_left
            odd_right = odd_tot - odd_left - arr[i]
            
        if even_left + odd_right == odd_left + even_right:
            ans += 1
            
        if i % 2 == 0: even_left += arr[i]
        else: odd_left += arr[i]
        
    print(ans)
solve()

File: Muc_Do_4_Rat_Kho/Bai20_XoaPhanTu/Solution.cpp
#include <bits/stdc++.h>
using namespace std;
// Bai 20: XoaPhanTu (LeetCode 1664: Ways to Make a Fair Array)
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI20.INP", "r")) {
        freopen("BAI20.INP", "r", stdin);
        freopen("BAI20.OUT", "w", stdout);
    }
    int n;
    if (!(cin >> n)) return 0;
    vector<long long> a(n);
    long long even_total = 0, odd_total = 0;
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
        if (i % 2 == 0) even_total += a[i];
        else odd_total += a[i];
    }
    
    long long even_left = 0, odd_left = 0;
    int ans = 0;
    for (int i = 0; i < n; ++i) {
        long long even_right = even_total - even_left - (i % 2 == 0 ? a[i] : 0);
        long long odd_right = odd_total - odd_left - (i % 2 != 0 ? a[i] : 0);
        
        if (even_left + odd_right == odd_left + even_right) {
            ans++;
        }
        
        if (i % 2 == 0) even_left += a[i];
        else odd_left += a[i];
    }
    cout << ans << "\n";
    return 0;
}
